pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -177,157 +178,159 @@ impl CustomScan for PdbScan {
                 return None;
             }
 
-            if let Some(quals) = quals {
-                let has_expressions = quals.contains_exprs();
-                let selectivity = if let Some(limit) = limit {
-                    // use the limit
-                    limit
-                        / table
-                            .reltuples()
-                            .map(|n| n as Cardinality)
-                            .unwrap_or(UNKNOWN_SELECTIVITY)
-                } else if restrict_info.len() == 1 {
-                    // we can use the norm_selec that already happened
-                    let norm_select = (*restrict_info.get_ptr(0).unwrap()).norm_selec;
-                    if norm_select != UNKNOWN_SELECTIVITY {
-                        norm_select
-                    } else {
-                        // assume PARAMETERIZED_SELECTIVITY
-                        PARAMETERIZED_SELECTIVITY
-                    }
+            if quals.is_none() && !maybe_needs_const_projections {
+                // if there are no qualifiers to apply, and no need for a score function, then
+                // don't use the custom scan
+                return None;
+            }
+
+            // at this point, we will execute the query, regardless of whether there were any Quals
+            // pushed down by the planner
+            let quals = quals.unwrap_or(Qual::All);
```

Comment:
Interesting.  #hmm 🤔

Reply:
Okay, so this answers my question from https://github.com/paradedb/paradedb/pull/2382#issuecomment-2764781755

This is as if the user had written

```sql
SEELCT *, paradedb.score(id) FROM table WHERE id @@@ paradedb.all();
```

In that case I think the score itself with either be `1` or `0` for every row.  If that's the effect we want, then could we instead keep `pdbscan/mod.rs` returning early if there's no Quals and just change `paradedb.score()` to return `1` (or `0`)?

Based on intuition, `WHERE id @@@ paradedb.all();` is going to be much slower than just a full sequential scan over the table (ie, no WHERE clause at all).

Reply:
Coming back to this...

From our offline discussions, this is actually incorrect.  Essentially, if `quals.is_none()` then we've already decided that we simply **cannot** answer the query with the custom scan.  Rewriting it to be `Qual::All` if we also think we maybe_needs_const_projections is incorrect as it changes the results Postgres would otherwise expect our scan to return.

Generally, I think this PR should be closed and we should examine the idea of teaching our custom scan to support Filters.  That discussion can happen offline.